pkg/reconciler/ksvc/resources/knative_configuration.go

```diff
@@ -10,69 +11,64 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func CreateKnativeConfiguration(kfsvc *v1alpha1.KFService) (*knservingv1alpha1.Configuration, *knservingv1alpha1.Configuration) {
+func CreateKnativeConfiguration(kfsvc *v1alpha1.KFService, name string, modelSpec *v1alpha1.ModelSpec) *knservingv1alpha1.Configuration {
+	if modelSpec == nil {
+		return nil
+	}
 	annotations := make(map[string]string)
-	if kfsvc.Spec.MinReplicas != 0 {
-		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(kfsvc.Spec.MinReplicas)
+	if modelSpec.MinReplicas != 0 {
+		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(modelSpec.MinReplicas)
+	}
+	if modelSpec.MaxReplicas != 0 {
+		annotations[autoscaling.MaxScaleAnnotationKey] = fmt.Sprint(modelSpec.MaxReplicas)
+	}
+
+	//User can pass down scaling target annotation to overwrite the target default 1
+	if _, ok := kfsvc.Annotations[autoscaling.TargetAnnotationKey]; !ok {
+		annotations[autoscaling.TargetAnnotationKey] = constants.DefaultScalingTarget
 	}
-	if kfsvc.Spec.MaxReplicas != 0 {
-		annotations[autoscaling.MaxScaleAnnotationKey] = fmt.Sprint(kfsvc.Spec.MaxReplicas)
+	//User can pass down scaling class annotation to overwrite the default scaling KPA
+	if _, ok := kfsvc.Annotations[autoscaling.ClassAnnotationKey]; !ok {
+		annotations[autoscaling.ClassAnnotationKey] = autoscaling.KPA
 	}
 
-	defaultConfiguration := &knservingv1alpha1.Configuration{
+	kfsvcAnnotations := filter(kfsvc.Annotations, isFilteredConfigurationAnnotation)
+
+	configuration := &knservingv1alpha1.Configuration{
```

Comment:
Shouldn't the annotations be applied to the configuration as well?

Reply:
? I think it does.

Reply:
Name:      name,
Namespace: kfsvc.Namespace,
Labels:    kfsvc.Labels,

Reply:
Oh, this is configuration's metadata, these annotations needs to be set on revision template's metadata if you check a few lines after that. I am not sure if configuration itself need those, I think that's how knative did too it passes the annotation from service to configuration's revision template.